src/solve.rs

```diff
@@ -53,16 +66,10 @@ pub async fn solve(
     let mut orders: Vec<(usize, OrderModel)> = orders.into_iter().map(|(i, y)| (i, y)).collect();
     // Filter out zero fee orders, as CowDexSolver is not good at matching liquidity orders.
     // Also they increase the revert risk, as Market Maker orders - at least the ones from zeroEx - can timing out and then cause simulation errors.
-    orders = orders
-        .into_iter()
-        .filter(|(_, order)| !is_zero_fee_order(order.clone()))
-        .collect();
-    // For simplicity, only solve for up to 10 orders
+    orders.retain(|(_, order)| !is_zero_fee_order(order.clone()));
+    // For simplicity, only solve for up to 4 orders
```

Comment:
We truncate further down the top 10 orders. Hence I think the comment was correct, but maybe not at the right place :) 

if you wanna do me a favor you update the comment to say that we filter for non_market orders, if there are more than 4 orders in the batch

Reply:
Oh wow, this logic is quite 🤡 though, isn't it? What if there is no market order in the batch, then we don't solve at all? But if there is 3 non-market maker ones, we do? And then we also cut at 10? Lots of non configurable magic values that affect the logic here in subtle ways 🤯